### ISSUE TITLE
react-compiler: BigInt literals are compiled to `undefined`

### ISSUE THREAD

**Issue by ZeroGDrive** [oxc-project/oxc#26161]
The React Compiler pass in `oxc-transform-react` replaces every **BigInt literal** inside a component or hook body with `undefined`. No diagnostic is emitted — `errors` is `[]` and `fatal` is `false`.

## Reproduction

```jsx
// min.jsx
export function Component() {
  const a = 1n;
  return <div>{String(a)}</div>;
}
```

```js
const { transformSync } = require('oxc-transform-react');
const fs = require('fs');
const r = transformSync('min.jsx', fs.readFileSync('min.jsx', 'utf8'));
console.log(r.code);
console.log('errors:', r.errors); // []
```

### Actual

```js
import { c as _c } from "react/compiler-runtime";
import { jsx as _jsx } from "react/jsx-runtime";
export function Component() {
	const $ = _c(1);
	let t0;
	if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /* @__PURE__ */ _jsx("div", { children: String(undefined) });
		//                                                ^^^^^^^^^ was `1n`
		$[0] = t0;
	} else {
		t0 = $[0];
	}
	return t0;
}
```

### Expected

The literal `1n` is preserved.

## The bug is in the React Compiler pass, not the parser or codegen

Same input, only the `reactCompiler` option differs:

```jsx
export function Component({ total }) {
  const remaining = total - 100n;
  const hasFee = total > 0n;
  return <div>{String(remaining)}{String(hasFee)}</div>;
}
```

<details>
<summary><code>reactCompiler: false</code> — correct</summary>

```js
import { jsxs as _jsxs } from "react/jsx-runtime";
export function Component({ total }) {
	const remaining = total - 100n;
	const hasFee = total > 0n;
	return /* @__PURE__ */ _jsxs("div", { children: [String(remaining), String(hasFee)] });
}
```

</details>

<details>
<summary><code>reactCompiler: true</code> (default) — both literals lost</summary>

```js
import { c as _c } from "react/compiler-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export function Component(t0) {
	const $ = _c(3);
	const { total } = t0;
	const remaining = total - undefined;
	const hasFee = total > undefined;
	...
}
```

</details>

For comparison, `babel-plugin-react-compiler@1.0.0` on the same input preserves `100n` and `0n`.

## Scope

- Affects BigInt **literals** (`0n`, `1n`, `100n`, …) in a component or hook body.
- The `BigInt(x)` **constructor call survives**, so values read from data look correct.
- Literals in plain non-component functions survive (that code is not transformed).

That combination makes it easy to miss: a codebase can look fine until a literal meets an operator.

```js
total - undefined   // TypeError / NaN
fee > undefined     // always false — the guard silently dies
paid ? total : 0n   // → paid ? total : undefined
```

## Versions

- `oxc-transform-react` **0.147.0** (also reproduced on 0.145.0)
- Node v24.18.1, macOS arm64

We hit this in production code doing money arithmetic with `bigint`, where the comparison case is silently wrong rather than throwing. We are staying on `babel-plugin-react-compiler` until this is fixed.
